core/trino-main/src/main/java/io/trino/execution/scheduler/BinPackingNodeAllocatorService.java

```diff
@@ -139,6 +142,7 @@ public BinPackingNodeAllocatorService(
         this.memoryRequirementIncreaseOnWorkerCrashEnabled = memoryRequirementIncreaseOnWorkerCrashEnabled;
         this.allowedNoMatchingNodePeriod = requireNonNull(allowedNoMatchingNodePeriod, "allowedNoMatchingNodePeriod is null");
         this.taskRuntimeMemoryEstimationOverhead = requireNonNull(taskRuntimeMemoryEstimationOverhead, "taskRuntimeMemoryEstimationOverhead is null");
+        this.eagerSpeculativeTasksNodeMemoryOvercommit = eagerSpeculativeTasksNodeMemoryOvercommit;
```

Comment:
We are able to over commit because we can always kill them if memory is not enough? Can we do the same for speculative tasks as well?

Reply:
We can. We can even for STANDARD, those will also be killed we run out of memory. But if we wanted to have more of STANDARD/SEPCULATIVE tasks running together we could just lower the memory estimate for those - would be more straightforward.
The starting memory estimate for tasks should be adjust to what level of concurrency we want - and that is why we picked 5GB which should match ~100GB worker node fine.

Overcommit for EAGER tasks is important because we believe those are "special". Typically they would not take much resources and have huge potential of finishing query early. And that is why we still want to run those even bumping taks concurrency a bit if cluster is fully booked on resources.